### ISSUE TITLE
Reduce hashing overhead

### ISSUE THREAD

**Issue by black-adder** [xichen2020/eventdb#143]
![image](https://user-images.githubusercontent.com/791791/53455513-cb7ab500-39f8-11e9-81db-a972702c346e.png)

The above is a flame graph captured whilst doing only writes. Hashing the field path is taking 14% of the time. We should look for a faster hash function or perform the hash during the parsing step #137 to improve write performance.

**Comment by notbdu** [comment]
Let's just switch it to use `xxHash`, we don't need a cryptographically secure hash fn.

**Comment by notbdu** [comment]
Benchmarks on  it's page look like we could get 2x perf gain. https://cyan4973.github.io/xxHash/

**Comment by xichen2020** [comment]
Yeah the reason why murmur3 was chosen was to provide a collision-free hash function so looking up the unhashable key (e.g., `[]byte`) is as fast as looking up the hash on the key. However there is theoretical possibility that there'll be collisions given enough keys, so we should change this to handle hashing properly. This requires regenerating the hash map with the proper hash+value comparisons, and using the right APIs to look up and set values.
